build.sbt

```diff
@@ -1,22 +1,105 @@
+import xerial.sbt.Sonatype._
+
 import com.scalapenos.sbt.prompt.SbtPrompt.autoImport._
 
 promptTheme := com.scalapenos.sbt.prompt.PromptThemes.ScalapenosTheme
 
-/** Project configurations */
+val commonSettings = Seq(
+  // sbt-git provides the isSnapshot.value variable, which will return true if there
+  // are no tags associated with the HEAD commit, or if there are uncommitted
+  // changes. However, sbt-git only appends the -SNAPSHOT suffix if there are
+  // uncommitted changes in the workspace.
+  //
+  // https://github.com/sbt/sbt-git/blob/f8caf9365be380cf101e9605af159b5e7f842d0c/src/main/scala/com/typesafe/sbt/SbtGit.scala#L173
+  version := {
+    // Avoid Cyclic reference involving error
+    if (git.gitCurrentTags.value.isEmpty || git.gitUncommittedChanges.value)
+      git.gitDescribedVersion.value.get + "-SNAPSHOT"
+    else
+      git.gitDescribedVersion.value.get
+  },
+  scalaVersion := "2.11.12",
+  crossScalaVersions := Seq("2.11.12", "2.12.8"),
+  resolvers ++= Seq(
+    Resolver.sonatypeRepo("releases"),
+    "locationtech-releases" at "https://repo.locationtech.org/content/groups/releases",
+    "locationtech-snapshots" at "https://repo.locationtech.org/content/groups/snapshots"
+  ),
+  addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
+  scalacOptions := Seq(
+    "-deprecation",
+    "-unchecked",
+    "-feature",
+    "-language:implicitConversions",
+    "-language:reflectiveCalls",
+    "-language:higherKinds",
+    "-language:postfixOps",
+    "-language:existentials",
+    "-language:experimental.macros",
+    "-feature",
+    "-Ypatmat-exhaust-depth", "100"
+  )
+)
+
+lazy val noPublishSettings = Seq(
+  publish := {},
+  publishLocal := {},
+  publishArtifact := false
+)
+
+lazy val publishSettings = Seq(
+  organization := "com.azavea.geotrellis",
+  organizationName := "GeoTrellis",
+  organizationHomepage := Some(new URL("https://geotrellis.io/")),
+  description := "MAML is used to create a declarative structure that describes a combination of map algebra operations.",
+  publishArtifact in Test := false
+) ++ sonatypeSettings ++ credentialSettings
+
+lazy val sonatypeSettings = Seq(
+  publishMavenStyle := true,
+
+  sonatypeProfileName := "com.azavea",
+  sonatypeProjectHosting := Some(GitHubHosting(user="geotrellis", repository="maml", email="systems@azavea.com")),
+  developers := List(
+    Developer(id = "moradology", name = "Nathan Zimmerman", email = "nzimmerman@azavea.com", url = url("https://github.com/moradology")),
+    Developer(id = "echeipesh", name = "Eugene Cheipesh", email = "echeipesh@azavea.com", url = url("https://github.com/echeipesh")),
+    Developer(id = "lossyrob", name = "Rob Emanuele", email = "remanuele@azavea.com", url = url("https://github.com/lossyrob"))
+  ),
+  licenses := Seq("Apache-2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+
+  publishTo := sonatypePublishTo.value
```

Comment:
There is logic behind here that checks if `version in ThisBuild` ends with `-SNAPSHOT` and determines the right repository to publish to.